File: backend/app/schemas/prompt_fragment.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class PromptFragmentBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=120)
    body: str = ""
    description: str | None = None


class PromptFragmentCreate(PromptFragmentBase):
    pass


class PromptFragmentUpdate(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=120)
    body: str | None = None
    description: str | None = None


class PromptFragmentOut(PromptFragmentBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/prompt-fragment.ts
export interface PromptFragment {
  id: number;
  name: string;
  body: string;
  description: string | null;
  tags_json: string | null;
  created_at: string;
  updated_at: string;
}

export type PromptFragmentCreate = Omit<
  PromptFragment,
  "id" | "created_at" | "updated_at"
>;

export type PromptFragmentUpdate = Partial<PromptFragmentCreate>;